cmd/main.go

```diff
@@ -1,74 +1,34 @@
 package main
 
 import (
-	"context"
 	"log"
 	"os"
 
 	"github.com/keptn-contrib/unleash-service/pkg/event_handler"
-
-	"github.com/kelseyhightower/envconfig"
-
-	cloudevents "github.com/cloudevents/sdk-go/v2"
-	"github.com/keptn/go-utils/pkg/lib/keptn"
+	"github.com/keptn/keptn/go-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
 )
 
-type envConfig struct {
-	// Port on which to listen for cloudevents
-	Port int    `envconfig:"RCV_PORT" default:"8080"`
-	Path string `envconfig:"RCV_PATH" default:"/"`
-}
+const getActionTriggeredEventType = "sh.keptn.event.action.triggered"
+const serviceName = "unleash-service"
+const envVarLogLevel = "LOG_LEVEL"
 
 func main() {
-	var env envConfig
-	if err := envconfig.Process("", &env); err != nil {
-		log.Fatalf("Failed to process env var: %s", err)
-	}
-
-	os.Exit(_main(os.Args[1:], env))
-}
-
-func _main(args []string, env envConfig) int {
-
-	ctx := context.Background()
-	ctx = cloudevents.WithEncodingStructured(ctx)
-
-	p, err := cloudevents.NewHTTP(cloudevents.WithPath(env.Path), cloudevents.WithPort(env.Port))
-	if err != nil {
-		log.Fatalf("failed to create client, %v", err)
-	}
-	c, err := cloudevents.NewClient(p)
-	if err != nil {
-		log.Fatalf("failed to create client, %v", err)
-	}
-	log.Fatal(c.StartReceiver(ctx, gotEvent))
-
-	return 0
-}
-
-func gotEvent(ctx context.Context, event cloudevents.Event) error {
-
-	var shkeptncontext string
-	_ = event.Context.ExtensionAs("shkeptncontext", &shkeptncontext)
-
-	logger := keptn.NewLogger(shkeptncontext, event.Context.GetID(), "unleash-service")
-
-	eventHandler, err := event_handler.NewEventHandler(event, logger)
-
-	if err != nil {
-		return err
-	}
-
-	if eventHandler == nil {
-		logger.Debug("No event handler registered for: " + event.Type())
-		return nil
-	}
-
-	err = eventHandler.HandleEvent()
-
-	if err != nil {
-		return err
+	if os.Getenv(envVarLogLevel) != "" {
+		logLevel, err := logrus.ParseLevel(os.Getenv(envVarLogLevel))
+		if err != nil {
+			logrus.WithError(err).Error("could not parse log level provided by 'LOG_LEVEL' env var")
+			logrus.SetLevel(logrus.InfoLevel)
+		} else {
+			logrus.SetLevel(logLevel)
+		}
 	}
 
-	return nil
+	log.Fatal(sdk.NewKeptn(
+		serviceName,
+		sdk.WithTaskHandler(
+			getActionTriggeredEventType,
+			event_handler.NewActionTriggeredHandler()),
```

Comment:
We can customize the event in the helm values, but the unleash-service only listens to the action triggered event. This can be problematic if the user changes the helm values. Maybe we should just remove the ability to customize the events in the helm values.